sdk/core/azure-mgmt-core/azure/mgmt/core/policies/_policy_token_header.py

```diff
@@ -0,0 +1,165 @@
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import json
+from typing import TypeVar, Any, Union, TYPE_CHECKING, Dict
+from azure.core.pipeline import PipelineRequest
+from azure.core.pipeline.transport import (
+    HttpResponse as LegacyHttpResponse,
+    HttpRequest as LegacyHttpRequest,
+)
+from azure.core.rest import HttpResponse, HttpRequest, AsyncHttpResponse
+from azure.core.pipeline.policies import SansIOHTTPPolicy
+from azure.core.exceptions import HttpResponseError
+from ..tools import parse_resource_id
+
+HTTPResponseType = TypeVar("HTTPResponseType", HttpResponse, LegacyHttpResponse)
+HTTPRequestType = TypeVar("HTTPRequestType", HttpRequest, LegacyHttpRequest)
+
+if TYPE_CHECKING:
+    from .._pipeline_client import ARMPipelineClient
+
+
+def _create_acquire_policy_request(request: PipelineRequest[HTTPRequestType]) -> HttpRequest:
+    """Create a request to acquire a policy token.
+
+    This method creates an HTTP request to the Azure Policy service to acquire
+    a policy token for external policy evaluation. The token will be used in
+    subsequent requests to enable policy evaluation.
+
+    :param request: The pipeline request for which to acquire a policy token
+    :type request: ~azure.core.pipeline.PipelineRequest
+    :return: The HTTP request to acquire policy token
+    :rtype: ~azure.core.rest.HttpRequest
+    :raises ~azure.core.exceptions.HttpResponseError: If subscription ID cannot be extracted from request URL
+    """
+    # try to get subscriptionId from request.http_request.url
+    try:
+        resource_info = parse_resource_id(request.http_request.url)
+        subscription_id = resource_info.get("subscription")
+    except Exception:  # pylint: disable=broad-except
+        subscription_id = None
+    if not subscription_id:
+        raise HttpResponseError("Failed to get subscriptionId from request url: {}".format(request.http_request.url))
```

Comment:
This shouldn't be an `HttpResponseError` - as we have no response yet. I don't even think we could call it an HttpRequestError...... and what circumstances would this happen? That would help us determine the best error to raise.

Reply:
The error occurs when users apply the policy for a http request which contains no `subsriptionId`. Since the policy needs subscriptionId to make a [http request call](https://github.com/Azure/azure-rest-api-specs/blob/c2f167ec164a2349235d8aee1b8189b883ec93c6/specification/resources/resource-manager/Microsoft.Authorization/stable/2025-03-01/policyTokens.json#L37) to get additional token, the policy has to raise exception when no subscriptionId can be used.
Maybe `SeviceRequestError` or just `AzureError`?